### DISCUSSION TITLE
Prohibit data uploading from certain parts of the experiment

### DISCUSSION THREAD

**Discussion by zhangchi199133** [jspsych/jsPsych#960] [Q&A] [answered]
Hi all,
In my experiment I have a procedure in which files are preloaded. I put this in the timeline using jsPsych.pluginAPI.preloadAudioFiles function. This resulted in that in the data sheet I collected there were a large chunk of data relating to file preloading, which is very much unwanted. I was wondering whether there are ways I can prohibit the data of file preloading from the final data? Thank you so much!
Best,
Chi

**Comment by jodeleeuw** [comment, marked as answer]
Hi Chi,

If you are using the `call-function` plugin the return value of the function is stored in the data. Can you change your function so that it doesn't return a large chunk? Perhaps `return null;` at the end of the function?

**Reply by zhangchi199133** [reply]
Hi Josh,

Thank you so much for the reply.

I tried to run the `call-function` with the `return plugin` changed to `return null`. But I got error code as follow, referring to the previous `fullscreen function`:

jspsych.js:899 Uncaught TypeError: Cannot read property 'info' of null
    at evaluateFunctionParameters (jspsych.js:899)
    at doTrial (jspsych.js:834)
    at nextTrial (jspsych.js:822)
    at Object.window.jsPsych.core.finishTrial (jspsych.js:267)
    at jspsych-fullscreen.js:95

Maybe I changed the wrong place? Thank you so much!

Best,

Chi

**Reply by vijaymarupudi** [reply]
You shouldn't change the plugin (I would revert that change), you should change the function you provided to the plugin in the `func` property to not return anything.

**Reply by zhangchi199133** [reply]
Hi there!

I changed added `return null` to the `func` of my `call-function`. But the data from manual preloading was still recorded. Please find below the codes I used.

```js
      var preload_audio_trial = {
            timeline_variables: trial_stimuli_bin1,
            timeline: [
              {
            type: "call-function",
            async: true,
            func: function (done) {
                jsPsych.getDisplayElement().innerHTML = '<p> Loading... </p>',
                jsPsych.pluginAPI.preloadAudioFiles([jsPsych.timelineVariable('audio', true)],
                function () {done({ preload: "success" })},
              );
              return null;
            }
          },
              {
            type: "call-function",
            async: true,
            func: function (done) {
                jsPsych.getDisplayElement().innerHTML = '<p> Loading... </p>',
                jsPsych.pluginAPI.preloadImages([jsPsych.timelineVariable('ppic', true)], function () {
                  done({ preload: "success" })
                });
                return null;
                }
              },
              {
            type: "call-function",
            async: true,
            func: function (done) {
                jsPsych.getDisplayElement().innerHTML = '<p> Loading... </p>',
                jsPsych.pluginAPI.preloadImages([jsPsych.timelineVariable('tpic', true)], function () {
                  done({ preload: "success" })
                });
                return null;
                }
              }
              ]
            };
```

Thanks a lot!

Chi

**Reply by jodeleeuw** [reply]
I think the issue here is that you are running a *separate* trial for each of the stimuli that you are preloading, so the jsPsych data will include a row for each stimulus. There's no way to avoid this in the current version of jsPsych. I'd recommend loading all of the stimuli in one function call instead.

**Reply by zhangchi199133** [reply]
OK! Thank you so much!
